api/src/context.rs

```diff
@@ -236,10 +239,111 @@ impl Context {
         address: AccountAddress,
         version: u64,
     ) -> Result<HashMap<StateKey, StateValue>> {
-        self.db
-            .get_state_values_by_key_prefix(&StateKeyPrefix::from(address), version)
+        let mut iter = self.db.get_prefixed_state_value_iterator(
+            &StateKeyPrefix::from(address),
+            None,
+            version,
+        )?;
+        let kvs = iter
+            .by_ref()
+            .take(MAX_REQUEST_LIMIT as usize)
+            .collect::<Result<_>>()?;
+        if iter.next().transpose()?.is_some() {
+            bail!("Too many state items under account ({:?}).", address);
+        }
+        Ok(kvs)
+    }
+
+    pub fn get_resources_by_pagination(
+        &self,
+        address: AccountAddress,
+        prev_state_key: Option<&StateKey>,
```

Comment:
is this expected from a restful interface input?

Reply:
What do you mean? Like the StateKey type? This function itself isn't the endpoint handler. In my PR on top of this one I add StateKeyWrapper that represents this as a hex string in the interface.